### PR TITLE
[7.x] Sync discovery_gce versions with main

### PR DESCRIPTION
**Pull request by jakelandis**
This commit syncs the dependencies for discovery GCE plugin with the dependency versions in use in main. 

Specifically, this commit upgrades the following for the discovery GCE plugin:

com.google.apis:google-api-services-compute:v1-rev160-1.23.0 -> v1-rev20220322-1.32
com.google.api-client:google-api-client:1.23.0 -> 1.33.1
com.google.oauth-client:google-oauth-client:1.23.0 -> 1.34.1
com.google.code.findbugs:jsr305:1.3.9 -> 3.0.2

and introduces the following for the discovery GCE plugin:

api "com.fasterxml.jackson.core:jackson-core:${versions.jackson}"
api "com.google.http-client:google-http-client-gson:1.41.1"
runtimeOnly 'com.google.guava:guava:32.0.1-jre'
runtimeOnly 'com.google.guava:failureaccess:1.0.1'
api 'io.opencensus:opencensus-api:0.30.0'
api 'io.opencensus:opencensus-contrib-http-util:0.30.0'
api 'io.grpc:grpc-context:1.27.2'

backport of https://github.com/elastic/elasticsearch/pull/85132 and https://github.com/elastic/elasticsearch/pull/91722

===

The combination of dependencies have been used in main for 6+ months and assuming CI agrees, this should be a safe dependency upgrade. 

diff of build.gradle against main : [diff_with_main.pdf](https://github.com/elastic/elasticsearch/files/13865265/diff_with_main.pdf) (main defines the version in the common versions file)

### CODE REVIEW

File: gradle/verification-metadata.xml [hunk 1]
```diff
@@ -464,6 +464,11 @@
             <sha256 value="357de0017d439c021a6a98d48e380a25a709dc9bc50e4b38c6ab84094e6f2e84" origin="Generated by Gradle"/>
          </artifact>
       </component>
+      <component group="com.google.api-client" name="google-api-client" version="1.33.1">
```
Comment (jakelandis) [inline gradle/verification-metadata.xml:467]: all changes here added via `./gradlew --write-verification-metadata sha256`

File: plugins/discovery-gce/build.gradle [hunk 2]
```diff
@@ -7,19 +7,30 @@ esplugin {
 }
 
 versions << [
-  'google': '1.23.0'
```
Comment (jakelandis) [inline plugins/discovery-gce/build.gradle:10]: changes copied and pasted from main : https://github.com/elastic/elasticsearch/blob/main/plugins/discovery-gce/build.gradle

**Comment by jakelandis** [comment]
@elasticmachine run elasticsearch-ci/packaging-tests-windows-sample 
@elasticmachine run elasticsearch-ci/windows-2019 / default-windows-archive-no-jdk / packaging-tests-windows-sample 

Those failed due to : 
Caused by: java.nio.file.FileSystemException: C:\bk\qa\os\build\testrun\destructiveDistroTest.default-windows-archive-no-jdk\temp\org.elasticsearch.packaging.util.Shell5816487018950537013.out: The process cannot access the file because it is being used by another process

I think that is unrelated here.

**Comment by jakelandis** [comment]
@elasticmachine run 'elasticsearch-ci/windows-2019 / default-windows-archive-no-jdk / packaging-tests-windows-sample'

**Review by arteam — APPROVED** [review]
LGTM

**Pull request merged**